applications/common/backend/src/main/java/pl/ds/kyanite/common/components/models/layouts/TemplateHandler.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2024 Dynamic Solutions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pl.ds.kyanite.common.components.models.layouts;
+
+/**
+ * This interface provides a common way of adding/overriding layouts for components.
+ * It would be nice to have in any component if there is a chance for its layout to be customized -
+ * consider it a contract.
+ *
+ * <p>
+ * Use cases:
+ * <ul>
+ *    <li>you are using/extending an existing component and you want to change its layout
+ *    or add a new one</li>
+ *    <li>you are developing a component that can have different layouts by design
+ *    (e.g. horizontal and vertical tabs)</li>
+ * </ul>
+ * </p>
+ *
+ * <p>Component must follow this rules:
+ * <ol>
+ *    <li>component has a default template under templates/template-default.html</li>
+ *    <li>component's model implements MultiTemplateComponent interface</li>
```

Comment:
Let's update the name in the javadoc

Reply:
I've updated the javadoc and hope the wording is a little clearer now.